xdsl/printer.py

```diff
@@ -380,6 +382,19 @@ def print_float(self, value: float, type: AnyFloat):
                     # default to full python precision
                     self.print_string(f"{repr(value)}")
 
+    def print_int(self, value: int, type: IntegerType | IndexType | None = None):
+        if (
+            isinstance(type, IntegerType)
+            and type.width.data == 1
+            and type.signedness.data == Signedness.SIGNLESS
+        ):
+            if value:
+                self.print_string("true")
+            else:
+                self.print_string("false")
+        else:
+            self.print_string(f"{value:d}")
+
```

Comment:
I think I would maybe rather just expose `print_int` without a type?
If we want to print an integer with a type, that usually means we already have an `IntegerAttr`, so we would use that.
Otherwise, we usually don't have a type? So I would just expose the `print_int` function without the type and move the "true"/"false" logic to `IntegerAttr`, what do you think?

Reply:
From the next PR, it seems I am mistaken.
My question is then why we would not print `true` or `false` with `Signedness.SIGN` and `Signedness.UNSIGNED`? (Maybe that's an MLIR thing).

I still think that maybe we should have a `print_int` and a `print_typed_int`?

Reply:
My aim was to try to unify the interfaces of `print_int` and `print_float`, though for `print_float` the type isn't optional anyways so maybe this doesn't work. As of this PR the printing for `IntegerAttr` uses `printer.print_int` (as I think it should). An alternative could be to restore the `type.print_without_type` functions but have these call (an untyped) `print_int`. What do you think?

Reply:
I don't think we need `type.print_without_type` tbh.
I think that's okay then that you pass the attribute, however, I would add a documentation comment that actually explains why you need the attribute (for printing as boolean)